### PR TITLE
Depend on oldest-supported-numpy instead of plain NumPy

### PR DESCRIPTION
**Pull request by ivilata**
Requiring an oldest version of NumPy causes less incompatibilities with other projects (esp. if they depend on oldest-supported-numpy themselves).

Tests seem to work just fine, and generated documentation still looks ok.

### CODE REVIEW

File: requirements-runtime.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
 msgpack
 ndindex>=1.4
-numpy>=1.20.3
+oldest-supported-numpy
```
Comment (FrancescAlted) [inline requirements-runtime.txt:3]: I think for runtime `numpy<2` is a better option (`oldest-supported-numpy` is mainly for building purposes).

File: requirements-test-wheels.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 pip
-numpy<2
+oldest-supported-numpy
```
Comment (FrancescAlted) [inline requirements-test-wheels.txt:2]: Ditto

File: requirements-tests.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 pip
-numpy<2
+oldest-supported-numpy
```
Comment (FrancescAlted) [inline requirements-tests.txt:2]: Ditto

**Review by FrancescAlted — CHANGES_REQUESTED** [review]

**Review by martaiborra — APPROVED** [review]
LGTM

**Comment by FrancescAlted** [comment]
I am doing the necessary changes for this in PR https://github.com/Blosc/python-blosc2/pull/135.  This can be closed now.

**Pull request closed without merging**